### PR TITLE
[IMP] mrp: revamp scss

### PR DESCRIPTION
**Pull request by stefanorigano**
This commit improves the design and removes unnecessary SCSS.
Part of the overall v16 SCSS optimization/restyle, task-2704984.

task-2896849

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by stefanorigano** [comment]
Hello @amoyaux , SCSS optimisations and an improved design, especially for bom report.
Could you have a look? Thanks!

cc. @adr-odoo , @pparidans

**Comment by stefanorigano** [comment]
>  we have already a task ongoing that redo the bom report from scratch. But we will reintroduce your design in it. 

Thanks @amoyaux .
Should I re-target the this to another branch? Or you guys will cherry-pick and handle it internally?

**Comment by amoyaux** [comment]
@stefanorigano You can merge this and we will handle the change on our branch

**Comment by stefanorigano** [comment]
> @stefanorigano You can merge this and we will handle the change on our branch

Ok, well... I don't have merge-rights :)

**Comment by amoyaux** [comment]
Okay I will just wait the review of @pparidans

**Comment by pparidans** [comment]
@stefanorigano As the Boostrap 5 PR has been finally merged, could you rebase this Pr one more (last?) time? Thx

**Comment by stefanorigano** [comment]
@pparidans , rebased and ready for tech. review.

File: addons/mrp/report/mrp_report_bom_structure.xml [hunk 1]
```diff
@@ -1,110 +1,105 @@
...
+                            <th class="text-start" groups="uom.group_uom">Unit of Measure</th>
+                            <th t-if="data['report_structure'] != 'bom_structure'" class="o_mrp_prod_cost text-end" title="This is the cost defined on the product.">Product Cost</th>
+                            <th t-if="data['report_structure'] != 'bom_structure'" class="o_mrp_bom_cost text-end" title="This is the cost based on the BoM of the product. It is computed by summing the costs of the components and operations needed to build the product.">BoM Cost</th>
+                            <th t-if="data['report_type'] == 'html' and data['has_attachments']" class="o_mrp_has_attachments" title="Files attached to the product">Attachments</th>
+                        </tr>
+                    </thead>
+                    <tbody>
+                        <tr>
+                            <td>
+                                <span><a href="#" t-if="data['report_type'] == 'html'" t-att-data-res-id="data['product'].id" t-att-data-model="'product.product'" class="o_mrp_bom_action"><t t-esc="data['bom_prod_name']"/></a><t t-else="" t-esc="data['bom_prod_name']"/></span>
```
Comment (adr-odoo) [inline addons/mrp/report/mrp_report_bom_structure.xml:35]: Could we take the opportunity to get rid of all of these <span>? Seems not necessary...

**Comment by clesgow** [comment]
Hello @stefanorigano, as @amoyaux said, I'm currently doing the rework of this report. I keep an eye on this PR so that your new design is properly implemented on this rework.
Up until your last rebase, I had a different PDF version of the report than yours, with seemingly less CSS used (h1 displayed as normal text, ...), and the result was quite different from what you had. Now when I test it on your runbot it seems you have the  same issue.
I honestly don't know if this is intended or not, nor if you intended to update the PDF version as well in the first place.

Cheers !

**Comment by stefanorigano** [comment]
> Now when I test it on your runbot it seems you have the same issue.

Hello @clesgow 👋
A friend told me that media-queries are not supported by `wkhtml` and BS5 we make use of them to generate  `<h1>` design.
Theoretically `wkhtml` should fallback to the default value, `font-size: calc(1.385rem + 1.62vw)`... but since it doesn’t support css custom-properties neither, at this point I would be very surprised if supports responsive units (**vw**)... it may be the reason why you get h1 displayed as normal test.

>  nor if you intended to update the PDF version as well in the first place.

The aim of the task was just to reduce the SCSS,  honestly I didn't compiled the PDF.


Not an expert in reports, but my suggestion is to create a (sort-of) `reboot.scss` file to provide the necessary fallback rules to `wkhtml`.
It should be injected in the right place and just before compilation.
Like `reset.min.css` I guess https://github.com/odoo/odoo/blob/master/addons/web/__manifest__.py#L298

cc. @amoyaux   , @adr-odoo

**Comment by stefanorigano** [comment]
Hello @adr-odoo , I applied the latest remarks.

**Comment by clesgow** [comment]
Thanks for your reply @stefanorigano, I'll look into it.
Glad to know why it was working differently on my end up until now !

**Comment by adr-odoo** [comment]
> Hello @adr-odoo , I applied the latest remarks.

There is still some unnecessary span and div I think..

**Comment by stefanorigano** [comment]
> > Hello @adr-odoo , I applied the latest remarks.
> 
> There is still some unnecessary span and div I think.. For me we can put the content directly in the td without wrapping it in a span..

Yes, a bit out of scope but no problem for me 👍

**Comment by stefanorigano** [comment]
Hello @adr-odoo , xml simplified as suggested. I kept the `<span/>` using `t-options` for obvious reasons.
`t-esc` replaced by `t-out` in all the affected lines.

@amoyaux , I took the occasion to fix the `report_bom_search` design.

cc @clesgow

File: addons/mrp/report/mrp_report_bom_structure.xml [hunk 2]
```diff
@@ -1,110 +1,106 @@
...
+                            <td t-if="data['report_type'] == 'html' and data['attachments']">
+                                <a t-if="data['attachments']" href="#" role="button" t-att-data-res-id="data['attachments'].ids" t-att-data-model="'mrp.document'" class="o_mrp_show_attachment_action fa fa-fw o_button_icon fa-files-o"/>
+                            </td>
+                        </tr>
+                        <t t-if="data['report_type'] == 'html'" t-call="mrp.report_mrp_bom_line"/>
+                        <t t-if="data['report_type'] == 'pdf'" t-call="mrp.report_mrp_bom_pdf_line"/>
+                    </tbody>
+                    <tfoot>
+                        <tr t-if="data['report_structure'] != 'bom_structure'" class="o_mrp_prod_cost">
+                            <td/>
```
Comment (adr-odoo) [inline addons/mrp/report/mrp_report_bom_structure.xml:58]: needed?

**Review by adr-odoo — APPROVED** [review]
@robodoo rebase-ff r+

**Pull request closed without merging**